### ISSUE TITLE
Saving to tape causes assertion failure

### ISSUE THREAD

**Issue by MBilderbeek** [openMSX/openMSX#986]
Scenario: save a file to tape.

Console: cassetteplayer new /tmp/test.wav

In my case, I used bsave to save some RAM to the wav file.

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  openmsx::CassettePlayer::updateTapePosition (this=this@entry=0x18fc480, 
    duration=duration@entry=..., time=...)
    at src/cassette/CassettePlayer.cc:449
449     assert(tapePos <= playImage->getEndTime());
[Current thread is 1 (Thread 0x7fa347cdb940 (LWP 6237))]
(gdb) bt full
#0  openmsx::CassettePlayer::updateTapePosition (this=this@entry=0x18fc480, duration=duration@entry=..., time=...)
    at src/cassette/CassettePlayer.cc:449
        __PRETTY_FUNCTION__ = "void openmsx::CassettePlayer::updateTapePosition(param, param)"
#1  0x0000000000464efd in openmsx::CassettePlayer::sync (this=0x18fc480, time=...) at src/cassette/CassettePlayer.cc:439
No locals.
#2  0x00000000004650f3 in openmsx::CassettePlayer::setSignal (this=0x18fc480, output=<optimized out>, time=...)
    at src/cassette/CassettePlayer.cc:430
No locals.
#3  0x0000000000435dd8 in openmsx::MSXPPI::writeC1 (this=0x18d0870, value=<optimized out>, time=...) at src/MSXPPI.cc:171
No locals.
#4  0x0000000000422e68 in openmsx::I8255::outputPortC (this=this@entry=0x18d0950, value=<optimized out>, time=..., time@entry=...)
    at src/I8255.cc:253
No locals.
#5  0x0000000000422ebf in openmsx::I8255::writeControlPort (this=0x18d0950, value=<optimized out>, time=...) at src/I8255.cc:285
        bitmask = <optimized out>
#6  0x00000000004abb20 in openmsx::MSXCPUInterface::writeIO (time=..., value=<optimized out>, port=<optimized out>, this=<optimized out>)
    at src/cpu/MSXCPUInterface.hh:121
No locals.
#7  openmsx::CPUCore<openmsx::Z80TYPE>::WRITE_PORT (cc=<optimized out>, value=<optimized out>, port=<optimized out>, 
    this=<optimized out>) at src/cpu/CPUCore.cc:584
No locals.
#8  openmsx::CPUCore<openmsx::Z80TYPE>::out_byte_a (this=<optimized out>) at src/cpu/CPUCore.cc:4020
No locals.
#9  openmsx::CPUCore<openmsx::Z80TYPE>::executeInstructions (this=0x15a2470) at src/cpu/CPUCore.cc:1167
        __PRETTY_FUNCTION__ = "void openmsx::CPUCore<CPU_POLICY>::executeInstructions() [with CPU_POLICY = openmsx::Z80TYPE]"
        opcodeTable = {0x4ae2af <openmsx::CPUCore<openmsx::Z80TYPE>::executeInstructions()+21647>, 
```

**Comment by MBilderbeek** [comment]
Looks a bit like #931?

**Comment by MBilderbeek** [comment]
Fixed by Wouter in commit a22edb171a823ec85d0cf8520b34538446f16a34

Thanks!
